instrumentation/netty/netty-4.1/library/src/main/java/io/opentelemetry/instrumentation/netty/v4_1/NettyClientTelemetryBuilder.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.netty.v4_1;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.netty.v4.common.internal.client.NettyClientInstrumenterFactory;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/** A builder of {@link NettyClientTelemetry}. */
+public final class NettyClientTelemetryBuilder {
+
+  private final OpenTelemetry openTelemetry;
+  private List<String> capturedRequestHeaders = Collections.emptyList();
+  private List<String> capturedResponseHeaders = Collections.emptyList();
+  private Map<String, String> peerServiceMapping = Collections.emptyMap();
+
+  NettyClientTelemetryBuilder(OpenTelemetry openTelemetry) {
+    this.openTelemetry = openTelemetry;
+  }
+
+  /**
+   * Configures the HTTP request headers that will be captured as span attributes.
+   *
+   * @param capturedRequestHeaders A list of HTTP header names.
+   */
+  @CanIgnoreReturnValue
+  public NettyClientTelemetryBuilder setCapturedRequestHeaders(
+      List<String> capturedRequestHeaders) {
+    this.capturedRequestHeaders = capturedRequestHeaders;
+    return this;
+  }
+
+  /**
+   * Configures the HTTP response headers that will be captured as span attributes.
+   *
+   * @param capturedResponseHeaders A list of HTTP header names.
+   */
+  @CanIgnoreReturnValue
+  public NettyClientTelemetryBuilder setCapturedResponseHeaders(
+      List<String> capturedResponseHeaders) {
+    this.capturedResponseHeaders = capturedResponseHeaders;
+    return this;
+  }
+
+  /**
+   * Configures the mapping between peer names and peer services used when capturing peer service
+   * span attributes.
+   *
+   * @param peerServiceMapping A map of peer names to peer services.
+   */
+  @CanIgnoreReturnValue
+  public NettyClientTelemetryBuilder setPeerServiceMapping(Map<String, String> peerServiceMapping) {
```

Comment:
Usually we're using the `addAttributesExtractor` method to provide the peer service mappings, e.g.:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/5184a523b6c851474e2c72db514670e274ad8c54/instrumentation/okhttp/okhttp-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/okhttp/v3_0/OkHttp3Singletons.java#L21-L23
I think that's because we wanted the library instrumentations to be more flexible about that; `PeerServiceAttributesExtractor` and using a `Map` with all possible peer service values are more for the javaagent use case.
WDYT about removing that method and allowing adding a custom `AttributesExtractor` instead?

Reply:
Yeah that sounds reasonable!